Vostok.Hosting/Components/Metrics/HealthCheckMetrics.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Linq;
+using Vostok.Hosting.Abstractions;
+using Vostok.Hosting.Abstractions.Diagnostics;
+using Vostok.Metrics;
+using Vostok.Metrics.Models;
+
+namespace Vostok.Hosting.Components.Metrics
+{
+    internal class HealthCheckMetrics : IObserver<HealthReport>
+    {
+        private readonly IMetricContext context;
+
+        private HealthCheckMetrics(IHealthTracker healthTracker, IMetricContext context)
+        {
+            this.context = context;
+
+            healthTracker.ObserveReports().Subscribe(this);
+        }
+
+        public static void Measure(IHealthTracker healthTracker, IVostokApplicationMetrics context)
+        {
+            // ReSharper disable once ObjectCreationAsStatement
+            new HealthCheckMetrics(healthTracker, context.Instance.WithTag(WellKnownTagKeys.Component, "HealthCheck"));
+        }
+
+        public void OnCompleted() {}
+
+        public void OnError(Exception error) {}
+
+        public void OnNext(HealthReport value)
+        {
+            foreach (var keyValuePair in value.Checks.Where(x => x.Value.Status != HealthStatus.Healthy))
+            {
+                context.Send(
+                    new MetricDataPoint(
+                        Convert.ToDouble(keyValuePair.Value.Status),
```

Comment:
Representing status as a numeric value is probably not the best idea due to user confusion. How does one infer that 1 means `Degraded` and 2 stands for `Failing` without studying the source code of `vostok.hosting.abstractions`?

It might be considerably easier to comprehend if status was just another tag. Like this:
`<identity>.HealthChecks.<name>.<status> = 1`

Also, come to think of it, we could probably benefit from a metric for overall instance health status (to enable less noisy alerts that could be investigated using per-check metrics).